### DISCUSSION TITLE
EventArgsConverter for generic events convertion

### DISCUSSION THREAD

**Discussion by dnviti** [CommunityToolkit/Maui#1133] [New Feature Discussions]
Should create a new EventArgsConverter to be used to catch generic EventArgs from any EventToCommandBehavior event and be able to pass it to the references Command like so:

```xaml
<ContentPage.Resources>
    <ResourceDictionary>
        <convs:EventArgsConverter x:Key="EventArgsConverter" />
    </ResourceDictionary>
</ContentPage.Resources>
...
<toolkit:EventToCommandBehavior
EventName="SelectionChanged"
EventArgsConverter="{StaticResource Key=EventArgsConverter}"
Command="{Binding SelectionChangedCommand}" />
```

And it's Command Usage

```cs
new Command<AnyEventArgsYouNeed>(async (args) => await Task.Delay(1));
```

**Comment by TheCodeTraveler** [comment]
Hi @dnviti!

I'm not sure I understand the scope of this new feature based on the example you've provided. 

Could you please further elaborate the intent of this feature and include its API design and a real example that uses the Converter.

**Comment by pictos** [comment]
Isn't this already implemented on the toolkit? 
https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/behaviors/event-to-command-behavior#1-use-the-generic-implementation

**Comment by CliffAgius** [comment]
Closed and answered
